### PR TITLE
Hotfix: use node fs-extra to handle copying and removing files #3hk00xt

### PR DESCRIPTION
**Pull request by hacktivist123**
This PR fixes this issue https://github.com/stackbit-themes/stackbit-examples/issues/17.  It introduces node [fs-extra](https://github.com/jprichardson/node-fs-extra) to handle the following:

- [x] Copying files on all OS types

- [x] Removing files on all OS types

- [x] Remove code where we are running commands directly

- [x] Increase performance a bit

### CODE REVIEW

**Comment by youvalv** [comment]
Task linked: [CU-3hk00xt `cp` Command doesn't work on Windows](https://app.clickup.com/t/3hk00xt)

File: index.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 import chalk from 'chalk';
 import { exec } from 'child_process';
-import fs from 'fs';
+import fs from 'fs-extra';
```
Comment (seancdavis) [inline index.js:5]: If you're going to use a package, it should be listed as a dependency in `package.json`. Although widely used, it's not safe to rely on dependencies' dependencies.

File: index.js [hunk 2]
```diff
@@ -39,10 +39,10 @@ async function installDependencies(dirName) {
 
 async function initGit(dirName) {
   console.log(`Setting up Git ...`);
-  await run(`rm -rf ${dirName}/.git`);
+  // remove .git folder
```
Comment (seancdavis) [inline index.js:42]: The code you wrote is clear. Comment is unnecessary.

File: index.js [hunk 3]
```diff
@@ -39,10 +39,10 @@ async function installDependencies(dirName) {
...
+  await fs.removeSync(`${dirName} / .git`);
```
Comment (seancdavis) [inline index.js:43]: Does this work? The syntax seems wrong. There should not be spaces in directory paths.

File: index.js (hunk 1 above)
Comment (seancdavis) [reply index.js:5]: I'd also like to know why this is necessary beyond the fs basics? It doesn't seem like we're doing anything fancy, and I like to keep dependencies to a minimum.

File: index.js [hunk 4]
```diff
@@ -148,18 +148,25 @@ async function cloneExample() {
     await run(`git clone --depth 1 --filter=blob:none --sparse ${config.examples.repoUrl} ${tmpDir}`);
     // Checkout just the example dir.
     await run(`cd ${tmpDir} && git sparse-checkout set ${args.example}`);
-    // Copy out into a new directory within current working directory.
-    await run(`cp -R ${tmpDir}/${args.example} ${dirName}`);
+
```
Comment (seancdavis) [inline index.js:151]: This is fine, but generally, when you're changing code, it's not necessary to adjust formatting, including adding new lines, unless there's a specific reason for doing it.

File: index.js [hunk 5]
```diff
@@ -148,18 +148,25 @@ async function cloneExample() {
...
+    // move out into a new directory.
+    await fs.moveSync(`${tmpDir}/${args.example}`, dirName);
```
Comment (seancdavis) [inline index.js:153]: Why move and not copy?

File: index.js [hunk 6]
```diff
@@ -148,18 +148,25 @@ async function cloneExample() {
...
-    if (fs.existsSync(dirName)) await run(`rm -rf ${dirName}`);
-    if (fs.existsSync(tmpDir)) await run(`rm -rf ${tmpDir}`);
+    if (fs.existsSync(dirName)) await fs.remove(dirName);
+    if (fs.existsSync(tmpDir))
+      // remove temp directory
+      await fs.remove(tmpDir, (err) => {
+        if (err) {
+          console.log(err);
+        }
+      });
```
Comment (seancdavis) [inline index.js:169]: This looks wrong. Using `await` and a callback? Also not sure why you want to be async here?

File: index.js [hunk 7]
```diff
@@ -148,18 +148,25 @@ async function cloneExample() {
...
     // Project Setup
     await installDependencies(dirName);
     await initGit(dirName);
   } catch (err) {
-    console.error(err);
-    if (fs.existsSync(dirName)) await run(`rm -rf ${dirName}`);
-    if (fs.existsSync(tmpDir)) await run(`rm -rf ${tmpDir}`);
+    if (fs.existsSync(dirName)) await fs.remove(dirName);
+    if (fs.existsSync(tmpDir))
+      // remove temp directory
```
Comment (seancdavis) [inline index.js:164]: Again, it's clear what you're doing below.

File: index.js [hunk 8]
```diff
@@ -148,18 +148,25 @@ async function cloneExample() {
...
+    await fs.removeSync(tmpDir);
 
     // Project Setup
     await installDependencies(dirName);
     await initGit(dirName);
   } catch (err) {
-    console.error(err);
-    if (fs.existsSync(dirName)) await run(`rm -rf ${dirName}`);
-    if (fs.existsSync(tmpDir)) await run(`rm -rf ${tmpDir}`);
+    if (fs.existsSync(dirName)) await fs.remove(dirName);
```
Comment (seancdavis) [inline index.js:162]: Why did you move/remove the error logging?

Comment (seancdavis) [reply index.js:162]: Also, FWIW, `run` was made into an async function so we didn't get into callback hell. If the fs lib has sync methods that don't use callbacks, I wouldn't go async here.

**Review by seancdavis — CHANGES_REQUESTED** [review]
Lots of comments. 

Please be aware of your formatting. When I first looked at this PR, nearly every line was changed for formatting.

I added a `.prettierrc` and updated this branch, so can now respect that file when making formatting updates.

File: index.js (hunk 3 above)
Comment (hacktivist123) [reply index.js:43]: This is because of my formatter, there shouldn't be any space. Thanks for enforcing the formatting.

File: index.js (hunk 4 above)
Comment (hacktivist123) [reply index.js:151]: This is my formatter's mistake, I enforced very specific rules. I'll update

File: index.js (hunk 5 above)
Comment (hacktivist123) [reply index.js:153]: Move is always faster than having to copy one by one then having to delete it in memory. When you move, you remove it from memory at that point but for copy, you'll need to do a clean up. 

This is a performance improvement.

File: index.js (hunk 8 above)
Comment (hacktivist123) [reply index.js:162]: Removing Error logging was a mistake,I forgot to put it back while trying to test something. My bad

File: index.js (hunk 1 above)
Comment (hacktivist123) [reply index.js:5]: It's because of the move and remove functions, there are some underlying issues especially with callbacks that fs has but fs-extra solves it.

**Review by hacktivist123 — COMMENTED** [review]
@seancdavis changes have been made based on your review

**Review by seancdavis — APPROVED** [review]
